miden-crypto/src/hash/algebraic_sponge/rescue/arch/mod.rs

```diff
@@ -73,13 +85,84 @@ pub mod optimized {
     ) -> bool {
         add_constants(state, ark);
         unsafe {
-            apply_inv_sbox(std::mem::transmute(state));
+            apply_inv_sbox(core::mem::transmute(state));
+        }
+        true
+    }
+
+    #[inline(always)]
+    pub fn add_constants_and_apply_ext_round(
+        state: &mut [Felt; STATE_WIDTH],
+        ark: &[Felt; STATE_WIDTH],
+    ) -> bool {
+        add_constants(state, ark);
+        unsafe {
+            apply_ext_round(core::mem::transmute(state));
         }
         true
     }
 }
 
-#[cfg(not(any(target_feature = "avx2", target_feature = "sve")))]
+// AVX512 OPTIMIZATIONS
+// ================================================================================================
+
+#[cfg(all(target_feature = "avx512f", target_feature = "avx512dq"))]
+mod x86_64_avx512;
+
+#[cfg(all(target_feature = "avx512f", target_feature = "avx512dq"))]
+pub mod optimized {
+    use super::x86_64_avx512::{apply_ext_round, apply_inv_sbox, apply_sbox};
+    use crate::{
+        Felt,
+        hash::algebraic_sponge::rescue::{STATE_WIDTH, add_constants},
+    };
+
+    #[inline(always)]
+    pub fn add_constants_and_apply_sbox(
+        state: &mut [Felt; STATE_WIDTH],
+        ark: &[Felt; STATE_WIDTH],
+    ) -> bool {
+        add_constants(state, ark);
+        unsafe {
+            apply_sbox(core::mem::transmute(state));
+        }
+        true
+    }
+
+    #[inline(always)]
+    pub fn add_constants_and_apply_inv_sbox(
+        state: &mut [Felt; STATE_WIDTH],
+        ark: &[Felt; STATE_WIDTH],
+    ) -> bool {
+        add_constants(state, ark);
+        unsafe {
+            apply_inv_sbox(core::mem::transmute(state));
+        }
+        true
+    }
+
+    #[inline(always)]
+    pub fn add_constants_and_apply_ext_round(
+        state: &mut [Felt; STATE_WIDTH],
+        ark: &[Felt; STATE_WIDTH],
+    ) -> bool {
+        add_constants(state, ark);
+        unsafe {
+            apply_ext_round(core::mem::transmute(state));
```

Comment:
Same transmute annotation warning here. Adding explicit type parameters would resolve the clippy warning: `core::mem::transmute::<&mut [BaseElement; 12], &mut [u64; 12]>(state)`.